pkg/cli/serve/config.go

```diff
@@ -11,29 +11,66 @@ import (
 // ReadFlagsFromFile checks for '@' prefix, if found try to read value from file.
 func ReadFlagsFromFile(cmd *cobra.Command, names ...string) error {
 	for _, name := range names {
-		value, err := cmd.Flags().GetString(name)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "unable to fetch value for key %s\n", name)
-			os.Exit(1)
-		}
-		if strings.HasPrefix(value, "@") {
+		flag := cmd.Flags().Lookup(name)
 
-			value = value[1:]
-			data, err := os.ReadFile(value)
+		switch flag.Value.Type() {
+		case "stringSlice":
+			values, err := cmd.Flags().GetStringSlice(name)
 			if err != nil {
-				return fmt.Errorf(
-					"can't read value of flag '%s' from file '%s': %w",
-					name, value, err,
-				)
+				fmt.Fprintf(os.Stderr, "unable to fetch value for key %s\n", name)
+				os.Exit(1)
+			}
+
+			if len(values) > 1 {
+				fmt.Fprintf(os.Stderr, "Must only specify one value when reading a string slice from file")
+				os.Exit(1)
 			}
-			value = strings.TrimSpace(string(data))
 
-			err = cmd.Flags().Set(name, value)
+			if strings.HasPrefix(values[0], "@") {
+				// Remove @ from start of the first word in string slice
+				values[0] = values[0][1:]
+				data, err := os.ReadFile(values[0])
+				if err != nil {
+					return fmt.Errorf(
+						"can't read value of flag '%s' from file '%s': %w",
+						name, values, err,
+					)
+				}
+
+				dataString := strings.TrimSpace(string(data))
+				values = strings.Split(string(dataString), ",")
```

Comment:
```suggestion
				values = strings.Split(dataString, ",")
```

Reply:
if we intent to use the `,` as the separator, not sure if we should document it somewhere